<div style="text-align: center;">
    <div style="background-color:#000000; padding-top:25px"> 
        <img src="netflix-logo.png" style="max-height:150px;" />
        <p style="color:#FFFFFF;
                   font-size:30px; 
                   font-weight:700; 
                   text-align: center; 
                   text-transform:uppercase;
                   letter-spacing:1px; 
                   line-height:1.2; 
                   padding-bottom:25px;">Exploratory Data Analysis</p>
    </div>
</div>

# Contents ⬇️ <a id='contents'></a>

[1. Contents ⬇️](#contents)   
[2. Introduction 📓](#introduction)  
[3. Project Goal 🎯](#project_goal)  
[4. Data Analysis 📊](#data-analysis)    
- [4.1 Initialization](#initialization)  
- [4.2 Load data](#load-data)  
- [4.3 Prepare the data](#prepare-the-data)    

# Introduction 📓 <a id='introduction'></a> 
[Back to Contents](#contents)

**Netflix Inc.** is an American media company based in Los Gatos, California. Founded in 1997 by Reed Hastings and Marc Randolph in Scotts Valley, California, it operates the **over-the-top subscription video on-demand service** Netflix brand, which includes **original films and television series commissioned or acquired by the company**, and third-party content licensed from other distributors.   

They have over 8000 movies or TV shows available on their platform, as of mid-2021, they have over 200M subscribers globally. This dataset has been taken from [Kaggle](https://www.kaggle.com/datasets/shivamb/netflix-shows) and consists of listings of all the movies and tv shows available on Netflix. This dataset was last updated in 2021.  

The dataset consists of 12 columns:
1. `show_id`: Unique ID for every Movie / TV show
2. `type`: Identifier - A Movie or TV Show
3. `title`: Title of the Movie / TV show
4. `director`: Director of the Movie
5. `cast`: Actors involved in the movie / TV show
6. `country`: Country where the movie / TV show was produced
7. `date_added`: Date it was added on Netflix
8. `release_year`: Actual Release year of the move / TV show
9. `rating`: TV Rating of the movie / TV show
10. `duration`: Total Duration - in minutes or number of seasons
11. `listed_in`: Genre of the movie / TV show
12. `description`: The summary description

# Project Goal 🎯 <a id='project_goal'></a>  
[Back to Contents](#contents)

# Data Analysis 📊 <a id='data-analysis'></a>  
[Back to Contents](#contents)

## Initialization <a id='initialization'></a>  
[Back to Contents](#contents)

To begin with, we need a few libraries for our statistical analysis - `pandas`, `matplotlib`, and `plotly.express`. We'll import all of them so that we can use the functions or methodds provided by them in our analysis:  

1. **Pandas**: It is a data manipulation library that provides functions to read, write and manipulate data in various formats.  

2. **Matplotlib**: It is a plotting library that is used to visualize data in various formats.  

3. **Plotly Express**: Plotly Express is a terse, consistent, high-level API for creating figures.

In [1]:
# Loading all the libraries
import pandas as pd
from matplotlib import pyplot as plt
from plotly import express as px

## Load data <a id='load-data'></a>  
[Back to Contents](#contents)

We have been data in `netflix_data.csv` file in the root directory in `csv` format with comma as field separator. We need to read the data file and load the data into DataFrame using `read_csv()` method provided by `pandas`.

In [2]:
# Load the data file into DataFrame
netflix_data = pd.read_csv('../netflix_data.csv')

## Prepare the data <a id='prepare-the-data'></a>  
[Back to Contents](#contents)

Let's get the general information of the data in the DataFrame - `netflix_data`:

In [3]:
# Print the general/summary information about the netflix_data DataFrame
netflix_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8807 entries, 0 to 8806
Data columns (total 12 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   show_id       8807 non-null   object
 1   type          8807 non-null   object
 2   title         8807 non-null   object
 3   director      6173 non-null   object
 4   cast          7982 non-null   object
 5   country       7976 non-null   object
 6   date_added    8797 non-null   object
 7   release_year  8807 non-null   int64 
 8   rating        8803 non-null   object
 9   duration      8804 non-null   object
 10  listed_in     8807 non-null   object
 11  description   8807 non-null   object
dtypes: int64(1), object(11)
memory usage: 825.8+ KB


The Dataframe - `netflix_data` has a total of **8807 rows and 12 columns**.

The **columns description** are as follows:
1. `show_id`: Unique ID for every Movie / TV show
2. `type`: Identifier - A Movie or TV Show
3. `title`: Title of the Movie / TV show
4. `director`: Director of the Movie
5. `cast`: Actors involved in the movie / TV show
6. `country`: Country where the movie / TV show was produced
7. `date_added`: Date it was added on Netflix
8. `release_year`: Actual Release year of the move / TV show
9. `rating`: TV Rating of the movie / TV show
10. `duration`: Total Duration - in minutes or number of seasons
11. `listed_in`: Genre of the movie / TV show
12. `description`: The summary description

There are some null values in `director`, `cast`, `country`, `date_added`, `rating` and `duration`. There are also some mismatch in data types of the columns that we'll address in a while.

**Let's get a sample of 10 rows from the Dataframe**:

In [4]:
# Print a sample of data for netflix_data
netflix_data.sample(n=10, random_state=100)

show_id     type                           title  \
3695   s3696  TV Show                Rabbids Invasion   
7338   s7339  TV Show  Los 10 años de Peter Capusotto   
6939   s6940    Movie      Haunting on Fraternity Row   
6626   s6627    Movie     Don't Be Afraid of the Dark   
3092   s3093  TV Show  Kevin Hart: Don’t F**k This Up   
5696   s5697    Movie                 1000 Rupee Note   
6472   s6473    Movie                      Chittagong   
4922   s4923    Movie                      Mercury 13   
4180   s4181    Movie                Old Lord Savanna   
684     s685  TV Show                           Elite   

                          director  \
3695                           NaN   
7338                           NaN   
6939                  Brant Sersen   
6626                    Troy Nixey   
3092                           NaN   
5696                Shrihari Sathe   
6472                Bedabrata Pain   
4922  David Sington, Heather Walsh   
4180                  André D'Elia   
684                            NaN   

                                                   cast  \
3695         Damien Laquet, David Gasman, Barbara Scaff   
7338                      Diego Capusotto, Ivana Acosta   
6939  Jacob Artist, Jayson Blair, Shanley Caswell, C...   
6626  Katie Holmes, Guy Pearce, Bailee Madison, Jack...   
3092                                                NaN   
5696  Usha Naik, Sandeep Pathak, Shrikant Yadav, Gan...   
6472  Manoj Bajpayee, Barry John, Delzad Hiwale, Veg...   
4922                                                NaN   
4180                                   Leonardo Ribeiro   
684   Danna Paola, Miguel Herrán, María Pedraza, Itz...   

                               country         date_added  release_year  \
3695                            France       July 1, 2019          2018   
7338                               NaN      June 29, 2018          2015   
6939                     United States      March 1, 2019          2018   
6626  United States, Australia, Mexico   November 2, 2019          2010   
3092                     United States  December 27, 2019          2019   
5696                             India   December 1, 2016          2014   
6472  United States, India, Bangladesh    January 1, 2018          2012   
4922                     United States     April 20, 2018          2018   
4180                            Brazil   January 18, 2019          2018   
684                              Spain      June 18, 2021          2021   

     rating   duration                                          listed_in  \
3695  TV-Y7   1 Season                              Kids' TV, TV Comedies   
7338  TV-MA   1 Season  International TV Shows, Spanish-Language TV Sh...   
6939  TV-MA     99 min                  Horror Movies, Independent Movies   
6626      R     99 min                                      Horror Movies   
3092  TV-MA   1 Season                                         Docuseries   
5696  TV-14     89 min                       Dramas, International Movies   
6472     NR    105 min   Dramas, Independent Movies, International Movies   
4922  TV-PG     79 min                                      Documentaries   
4180  TV-14     96 min                Documentaries, International Movies   
684   TV-MA  4 Seasons  Crime TV Shows, International TV Shows, Spanis...   

                                            description  
3695  Giant, rambunctious rabbits have invaded and a...  
7338  Fictional personality Peter Capusotto parodies...  
6939  When a fraternity's last big luau serves up ho...  
6626  Young Sally Hurst discovers she isn't alone in...  
3092  Amid turmoil in his career and marriage, comed...  
5696  After randomly receiving a handsome political ...  
6472  In the turbulent 1930s of British colonial Ind...  
4922  After rigorous testing in 1961, a small group ...  
4180  This documentary captures the environmental an...  
684   When three working-class teens enroll in an ex..

If we see the data above, we could say that `date_added` and `duration` columns should have correct data types for better analysis. We'll fix that in a while.

We can use the `duplicated()` method together with `sum()` to **check if we have any duplicate rows in the DataFrame - `netflix_data`**. `duplicated()` method returns a boolean Series (True/False) denoting duplicate rows. So, we could apply `sum()` over that series to get a summation of all the True(s) - and False(s).

In [5]:
# Checking for duplicated records in netflix data
netflix_data.duplicated().sum()

0

**We don't have any duplicate rows in the `netflix_data` Dataframe**.  

Let's check for just duplicate show IDs using `duplicated()` method together with `sum()`. Since, this time we want to find out if we have any duplicate show IDs, we will first get a Series of data for `show_id` column and then, apply `duplicated()` method along with `sum()` on it.

In [6]:
# Checking for just duplicate show IDs
netflix_data['show_id'].duplicated().sum()

0

**We don't have any duplicate show IDs in the `netflix_data` Dataframe**.

Let's verify that the `type` column contains values for only the two type of shows - **Movie** and **TV Show**:

In [7]:
# Check type column contains only two values - Movie and TV Show
netflix_data['type'].unique()

array(['Movie', 'TV Show'], dtype=object)

**The `type` column has correct values and contains values for only the two type of shows - Movie and TV Show**.

Let's verify that the `title` column doesn't have any duplicates:

In [8]:
netflix_data['title'].duplicated().sum()

0

Great! So, we don't have any duplicate movie titles in the dataset.

### Fix Data

We know from before that there are some null values in `director`, `cast`, `country`, `date_added`, `rating` and `duration` columns. 

**Let's see how many missing values do we have, by using `isna().sum()` in `director` column.** 

The `isna()` method returns a Series (or Dataframe) containing Boolean values - True or False, indicating whether the values in the cell is missing or not. The missing values that are considered by the `isna()` method are - `NaN` in numeric arrays, `None` or `NaN` in object arrays, `NaT` in datetimelike. It doesn't take any other user-defined values into account.

The `sum()` method adds up all the Boolean values by `isna()` method and gives us the total count of missing values.

In [9]:
# Counting the number of missing values in the director column
netflix_data['director'].isna().sum()

2634

**There are in total 2634 missing values in the `director` column**.  

Let's print the records where `director` column has missing values to peek into the data:

In [10]:
netflix_data[netflix_data['director'].isna()].head(10)

show_id     type                                title director  \
1       s2  TV Show                        Blood & Water      NaN   
3       s4  TV Show                Jailbirds New Orleans      NaN   
4       s5  TV Show                         Kota Factory      NaN   
10     s11  TV Show  Vendetta: Truth, Lies and The Mafia      NaN   
14     s15  TV Show      Crime Stories: India Detectives      NaN   
15     s16  TV Show                    Dear White People      NaN   
17     s18  TV Show                      Falsa identidad      NaN   
19     s20  TV Show                               Jaguar      NaN   
21     s22  TV Show               Resurrection: Ertugrul      NaN   
25     s26  TV Show                 Love on the Spectrum      NaN   

                                                 cast        country  \
1   Ama Qamata, Khosi Ngema, Gail Mabalane, Thaban...   South Africa   
3                                                 NaN            NaN   
4   Mayur More, Jitendra Kumar, Ranjan Raj, Alam K...          India   
10                                                NaN            NaN   
14                                                NaN            NaN   
15  Logan Browning, Brandon P. Bell, DeRon Horton,...  United States   
17  Luis Ernesto Franco, Camila Sodi, Sergio Goyri...         Mexico   
19  Blanca Suárez, Iván Marcos, Óscar Casas, Adriá...            NaN   
21  Engin Altan Düzyatan, Serdar Gökhan, Hülya Dar...         Turkey   
25                                   Brooke Satchwell      Australia   

            date_added  release_year rating   duration  \
1   September 24, 2021          2021  TV-MA  2 Seasons   
3   September 24, 2021          2021  TV-MA   1 Season   
4   September 24, 2021          2021  TV-MA  2 Seasons   
10  September 24, 2021          2021  TV-MA   1 Season   
14  September 22, 2021          2021  TV-MA   1 Season   
15  September 22, 2021          2021  TV-MA  4 Seasons   
17  September 22, 2021          2020  TV-MA  2 Seasons   
19  September 22, 2021          2021  TV-MA   1 Season   
21  September 22, 2021          2018  TV-14  5 Seasons   
25  September 21, 2021          2021  TV-14  2 Seasons   

                                            listed_in  \
1     International TV Shows, TV Dramas, TV Mysteries   
3                              Docuseries, Reality TV   
4   International TV Shows, Romantic TV Shows, TV ...   
10  Crime TV Shows, Docuseries, International TV S...   
14       British TV Shows, Crime TV Shows, Docuseries   
15                             TV Comedies, TV Dramas   
17  Crime TV Shows, Spanish-Language TV Shows, TV ...   
19  International TV Shows, Spanish-Language TV Sh...   
21  International TV Shows, TV Action & Adventure,...   
25     Docuseries, International TV Shows, Reality TV   

                                          description  
1   After crossing paths at a party, a Cape Town t...  
3   Feuds, flirtations and toilet talk go down amo...  
4   In a city of coaching centers known to train I...  
10  Sicily boasts a bold "Anti-Mafia" coalition. B...  
14  Cameras following Bengaluru police on the job ...  
15  Students of color navigate the daily slights a...  
17  Strangers Diego and Isabel flee their home in ...  
19  In the 1960s, a Holocaust survivor joins a gro...  
21  When a good deed unwittingly endangers his cla...  
25  Finding love can be hard for anyone. For young...

Since, the records where `director` column has missing values, has useful information, we can't just drop the records. So, we will replace the missing values in `director` column with `Unknown`:

In [11]:
# Filling missing values in director column with 'Unknown'
netflix_data['director'] = netflix_data['director'].fillna('Unknown')

Let's count the number of missing values in the `director` column once again:

In [12]:
# Counting the number of missing values in the director column
netflix_data['director'].isna().sum()

0

Awesome! Let's see how many missing values do we have, by using `isna().sum()` in `cast` column.

In [13]:
# Counting the number of missing values in the cast column
netflix_data['cast'].isna().sum()

825

**There are in total 825 missing values in the `cast` column**.  

Let's print the records where `cast` column has missing values to peek into the data:

In [14]:
netflix_data[netflix_data['cast'].isna()].head(10)

show_id     type                                              title  \
0       s1    Movie                               Dick Johnson Is Dead   
3       s4  TV Show                              Jailbirds New Orleans   
10     s11  TV Show                Vendetta: Truth, Lies and The Mafia   
14     s15  TV Show                    Crime Stories: India Detectives   
16     s17    Movie  Europe's Most Dangerous Man: Otto Skorzeny in ...   
20     s21  TV Show    Monsters Inside: The 24 Faces of Billy Milligan   
45     s46    Movie                             My Heroes Were Cowboys   
66     s67  TV Show                      Raja Rasoi Aur Anya Kahaniyan   
69     s70  TV Show                     Stories by Rabindranath Tagore   
74     s75  TV Show          The World's Most Amazing Vacation Rentals   

                                         director cast        country  \
0                                 Kirsten Johnson  NaN  United States   
3                                         Unknown  NaN            NaN   
10                                        Unknown  NaN            NaN   
14                                        Unknown  NaN            NaN   
16  Pedro de Echave García, Pablo Azorín Williams  NaN            NaN   
20                                Olivier Megaton  NaN            NaN   
45                                    Tyler Greco  NaN            NaN   
66                                        Unknown  NaN          India   
69                                        Unknown  NaN          India   
74                                        Unknown  NaN            NaN   

            date_added  release_year rating   duration  \
0   September 25, 2021          2020  PG-13     90 min   
3   September 24, 2021          2021  TV-MA   1 Season   
10  September 24, 2021          2021  TV-MA   1 Season   
14  September 22, 2021          2021  TV-MA   1 Season   
16  September 22, 2021          2020  TV-MA     67 min   
20  September 22, 2021          2021  TV-14   1 Season   
45  September 16, 2021          2021     PG     23 min   
66  September 15, 2021          2014   TV-G   1 Season   
69  September 15, 2021          2015  TV-PG   1 Season   
74  September 14, 2021          2021  TV-PG  2 Seasons   

                                            listed_in  \
0                                       Documentaries   
3                              Docuseries, Reality TV   
10  Crime TV Shows, Docuseries, International TV S...   
14       British TV Shows, Crime TV Shows, Docuseries   
16                Documentaries, International Movies   
20  Crime TV Shows, Docuseries, International TV S...   
45                                      Documentaries   
66                 Docuseries, International TV Shows   
69                  International TV Shows, TV Dramas   
74                                         Reality TV   

                                          description  
0   As her father nears the end of his life, filmm...  
3   Feuds, flirtations and toilet talk go down amo...  
10  Sicily boasts a bold "Anti-Mafia" coalition. B...  
14  Cameras following Bengaluru police on the job ...  
16  Declassified documents reveal the post-WWII li...  
20  In the late 1970s, an accused serial rapist cl...  
45  Robin Wiltshire's painful childhood was rescue...  
66  Explore the history and flavors of regional In...  
69  The writings of Nobel Prize winner Rabindranat...  
74  With an eye for every budget, three travelers ...

Since, the records where `cast` column has missing values, has useful information, we can't just drop the records. So, we will replace the missing values in `cast` column with `Unknown`:

In [15]:
# Filling missing values in cast column with 'Unknown'
netflix_data['cast'] = netflix_data['cast'].fillna('Unknown')

Let's count the number of missing values in the `cast` column once again:

In [16]:
# Counting the number of missing values in the cast column
netflix_data['cast'].isna().sum()

0

Awesome! Let's see how many missing values do we have, by using `isna().sum()` in `country` column.

In [17]:
# Counting the number of missing values in the country column
netflix_data['country'].isna().sum()

831

**There are in total 831 missing values in the `country` column**.  

Let's print the records where `country` column has missing values to peek into the data:

In [18]:
netflix_data[netflix_data['country'].isna()].head(10)

show_id     type                                              title  \
2       s3  TV Show                                          Ganglands   
3       s4  TV Show                              Jailbirds New Orleans   
5       s6  TV Show                                      Midnight Mass   
6       s7    Movie                   My Little Pony: A New Generation   
10     s11  TV Show                Vendetta: Truth, Lies and The Mafia   
11     s12  TV Show                                   Bangkok Breaking   
13     s14    Movie                   Confessions of an Invisible Girl   
14     s15  TV Show                    Crime Stories: India Detectives   
16     s17    Movie  Europe's Most Dangerous Man: Otto Skorzeny in ...   
18     s19    Movie                                          Intrusion   

                                         director  \
2                                 Julien Leclercq   
3                                         Unknown   
5                                   Mike Flanagan   
6                   Robert Cullen, José Luis Ucha   
10                                        Unknown   
11                              Kongkiat Komesiri   
13                                  Bruno Garotti   
14                                        Unknown   
16  Pedro de Echave García, Pablo Azorín Williams   
18                                     Adam Salky   

                                                 cast country  \
2   Sami Bouajila, Tracy Gotoas, Samuel Jouy, Nabi...     NaN   
3                                             Unknown     NaN   
5   Kate Siegel, Zach Gilford, Hamish Linklater, H...     NaN   
6   Vanessa Hudgens, Kimiko Glenn, James Marsden, ...     NaN   
10                                            Unknown     NaN   
11  Sukollawat Kanarot, Sushar Manaying, Pavarit M...     NaN   
13  Klara Castanho, Lucca Picon, Júlia Gomes, Marc...     NaN   
14                                            Unknown     NaN   
16                                            Unknown     NaN   
18  Freida Pinto, Logan Marshall-Green, Robert Joh...     NaN   

            date_added  release_year rating  duration  \
2   September 24, 2021          2021  TV-MA  1 Season   
3   September 24, 2021          2021  TV-MA  1 Season   
5   September 24, 2021          2021  TV-MA  1 Season   
6   September 24, 2021          2021     PG    91 min   
10  September 24, 2021          2021  TV-MA  1 Season   
11  September 23, 2021          2021  TV-MA  1 Season   
13  September 22, 2021          2021  TV-PG    91 min   
14  September 22, 2021          2021  TV-MA  1 Season   
16  September 22, 2021          2020  TV-MA    67 min   
18  September 22, 2021          2021  TV-14    94 min   

                                            listed_in  \
2   Crime TV Shows, International TV Shows, TV Act...   
3                              Docuseries, Reality TV   
5                  TV Dramas, TV Horror, TV Mysteries   
6                            Children & Family Movies   
10  Crime TV Shows, Docuseries, International TV S...   
11  Crime TV Shows, International TV Shows, TV Act...   
13                 Children & Family Movies, Comedies   
14       British TV Shows, Crime TV Shows, Docuseries   
16                Documentaries, International Movies   
18                                          Thrillers   

                                          description  
2   To protect his family from a powerful drug lor...  
3   Feuds, flirtations and toilet talk go down amo...  
5   The arrival of a charismatic young priest brin...  
6   Equestria's divided. But a bright-eyed hero be...  
10  Sicily boasts a bold "Anti-Mafia" coalition. B...  
11  Struggling to earn a living in Bangkok, a man ...  
13  When the clever but socially-awkward Tetê join...  
14  Cameras following Bengaluru police on the job ...  
16  Declassified documents reveal the post-WWII li...  
18  After a deadly home invasion at a couple’s new...

Since, the records where `country` column has missing values, has useful information, we can't just drop the records. So, we will replace the missing values in `country` column with `Unknown`:

In [19]:
# Filling missing values in country column with 'Unknown'
netflix_data['country'] = netflix_data['country'].fillna('Unknown')

Let's count the number of missing values in the `country` column once again:

In [20]:
# Counting the number of missing values in the country column
netflix_data['country'].isna().sum()

0

Awesome! Let's see how many missing values do we have, by using `isna().sum()` in `rating` column.

In [21]:
# Counting the number of missing values in the rating column
netflix_data['rating'].isna().sum()

4

**There are in total 4 missing values in the `rating` column**.  

Let's print the records where `rating` column has missing values to peek into the data:

In [22]:
netflix_data[netflix_data['rating'].isna()]

show_id     type                                              title  \
5989   s5990    Movie  13TH: A Conversation with Oprah Winfrey & Ava ...   
6827   s6828  TV Show                  Gargantia on the Verdurous Planet   
7312   s7313  TV Show                                       Little Lunch   
7537   s7538    Movie                               My Honor Was Loyalty   

             director                                               cast  \
5989          Unknown                        Oprah Winfrey, Ava DuVernay   
6827          Unknown  Kaito Ishikawa, Hisako Kanemoto, Ai Kayano, Ka...   
7312          Unknown  Flynn Curry, Olivia Deeble, Madison Lu, Oisín ...   
7537  Alessandro Pepe  Leone Frisa, Paolo Vaccarino, Francesco Miglio...   

        country        date_added  release_year rating  duration  \
5989    Unknown  January 26, 2017          2017    NaN    37 min   
6827      Japan  December 1, 2016          2013    NaN  1 Season   
7312  Australia  February 1, 2018          2015    NaN  1 Season   
7537      Italy     March 1, 2017          2015    NaN   115 min   

                                 listed_in  \
5989                                Movies   
6827  Anime Series, International TV Shows   
7312                 Kids' TV, TV Comedies   
7537                                Dramas   

                                            description  
5989  Oprah Winfrey sits down with director Ava DuVe...  
6827  After falling through a wormhole, a space-dwel...  
7312  Adopting a child's perspective, this show take...  
7537  Amid the chaos and horror of World War II, a c...

Since, the records where `rating` column has missing values, has useful information, we can't just drop the records. So, we will replace the missing values in `rating` column with `Unknown`:

In [23]:
# Filling missing values in rating column with 'Unknown'
netflix_data['rating'] = netflix_data['rating'].fillna('Unknown')

Let's count the number of missing values in the `rating` column once again:

In [24]:
# Counting the number of missing values in the rating column
netflix_data['rating'].isna().sum()

0

Let's see what are the set of values for `rating` column in the Dataframe - `netflix_data` to validate that they have meaningful values:

In [25]:
netflix_data['rating'].unique()

array(['PG-13', 'TV-MA', 'PG', 'TV-14', 'TV-PG', 'TV-Y', 'TV-Y7', 'R',
       'TV-G', 'G', 'NC-17', '74 min', '84 min', '66 min', 'NR',
       'Unknown', 'TV-Y7-FV', 'UR'], dtype=object)

It seems all the values are valid except the ones with ratings - `74 min`, `84 min` and `66 min`. These values doesn't seem to be valid values for ratings. Let's inspect the record where we have either of these three ratings:

In [26]:
netflix_data[(netflix_data['rating'] == '74 min') | (netflix_data['rating'] == '84 min') | (netflix_data['rating'] == '66 min') ]

show_id   type                                 title    director  \
5541   s5542  Movie                       Louis C.K. 2017  Louis C.K.   
5794   s5795  Movie                 Louis C.K.: Hilarious  Louis C.K.   
5813   s5814  Movie  Louis C.K.: Live at the Comedy Store  Louis C.K.   

            cast        country          date_added  release_year  rating  \
5541  Louis C.K.  United States       April 4, 2017          2017  74 min   
5794  Louis C.K.  United States  September 16, 2016          2010  84 min   
5813  Louis C.K.  United States     August 15, 2016          2015  66 min   

     duration listed_in                                        description  
5541      NaN    Movies  Louis C.K. muses on religion, eternal love, gi...  
5794      NaN    Movies  Emmy-winning comedy writer Louis C.K. brings h...  
5813      NaN    Movies  The comic puts his trademark hilarious/thought...

It seems that the values for the `rating` is supposed to be the values for the `duration` in the Dataframe - `netflix_data` for these 3 records. Let's fix that first:

In [27]:
# Fix row one with show_id - s5542
netflix_data.loc[netflix_data['show_id'] == 's5542', 'duration'] = '74 min'
netflix_data.loc[netflix_data['show_id'] == 's5542', 'rating'] = 'Unknown'
netflix_data[netflix_data['show_id'] == 's5542']

show_id   type            title    director        cast        country  \
5541   s5542  Movie  Louis C.K. 2017  Louis C.K.  Louis C.K.  United States   

         date_added  release_year   rating duration listed_in  \
5541  April 4, 2017          2017  Unknown   74 min    Movies   

                                            description  
5541  Louis C.K. muses on religion, eternal love, gi...

In [28]:
# Fix row one with show_id - s5795
netflix_data.loc[netflix_data['show_id'] == 's5795', 'duration'] = '84 min'
netflix_data.loc[netflix_data['show_id'] == 's5795', 'rating'] = 'Unknown'
netflix_data[netflix_data['show_id'] == 's5795']

show_id   type                  title    director        cast  \
5794   s5795  Movie  Louis C.K.: Hilarious  Louis C.K.  Louis C.K.   

            country          date_added  release_year   rating duration  \
5794  United States  September 16, 2016          2010  Unknown   84 min   

     listed_in                                        description  
5794    Movies  Emmy-winning comedy writer Louis C.K. brings h...

In [29]:
# Fix row one with show_id - s5814
netflix_data.loc[netflix_data['show_id'] == 's5814', 'duration'] = '66 min'
netflix_data.loc[netflix_data['show_id'] == 's5814', 'rating'] = 'Unknown'
netflix_data[netflix_data['show_id'] == 's5814']

show_id   type                                 title    director  \
5813   s5814  Movie  Louis C.K.: Live at the Comedy Store  Louis C.K.   

            cast        country       date_added  release_year   rating  \
5813  Louis C.K.  United States  August 15, 2016          2015  Unknown   

     duration listed_in                                        description  
5813   66 min    Movies  The comic puts his trademark hilarious/thought...

Awesome! Let's now convert the datatype of `date_added` column from **String** to **DateTime** of the format - `YYYY-MM-DD`:

In [30]:
# Convert date_added to datetime format
netflix_data['date_added'] = pd.to_datetime(netflix_data['date_added'].str.strip())

Let's have a peek into the data after changing the datatype of `date_added` column:

In [31]:
netflix_data.head(5)

show_id     type                  title         director  \
0      s1    Movie   Dick Johnson Is Dead  Kirsten Johnson   
1      s2  TV Show          Blood & Water          Unknown   
2      s3  TV Show              Ganglands  Julien Leclercq   
3      s4  TV Show  Jailbirds New Orleans          Unknown   
4      s5  TV Show           Kota Factory          Unknown   

                                                cast        country  \
0                                            Unknown  United States   
1  Ama Qamata, Khosi Ngema, Gail Mabalane, Thaban...   South Africa   
2  Sami Bouajila, Tracy Gotoas, Samuel Jouy, Nabi...        Unknown   
3                                            Unknown        Unknown   
4  Mayur More, Jitendra Kumar, Ranjan Raj, Alam K...          India   

  date_added  release_year rating   duration  \
0 2021-09-25          2020  PG-13     90 min   
1 2021-09-24          2021  TV-MA  2 Seasons   
2 2021-09-24          2021  TV-MA   1 Season   
3 2021-09-24          2021  TV-MA   1 Season   
4 2021-09-24          2021  TV-MA  2 Seasons   

                                           listed_in  \
0                                      Documentaries   
1    International TV Shows, TV Dramas, TV Mysteries   
2  Crime TV Shows, International TV Shows, TV Act...   
3                             Docuseries, Reality TV   
4  International TV Shows, Romantic TV Shows, TV ...   

                                         description  
0  As her father nears the end of his life, filmm...  
1  After crossing paths at a party, a Cape Town t...  
2  To protect his family from a powerful drug lor...  
3  Feuds, flirtations and toilet talk go down amo...  
4  In a city of coaching centers known to train I...

Let's also derive a new column - `year_added` from `date_added`:

In [32]:
netflix_data['year_added'] = netflix_data['date_added'].dt.strftime('%Y')

Let's see what are the years we have in `year_added` column:

In [33]:
netflix_data['year_added'].unique()

array(['2021', '2020', '2019', '2018', '2017', '2016', '2015', '2014',
       '2013', '2012', '2011', '2009', '2008', nan, '2010'], dtype=object)

We've missing values too in the `year_added` column. Let's replace it with `Unknown`:

In [34]:
# Filling missing values in year_added column with 'Unknown'
netflix_data['year_added'] = netflix_data['year_added'].fillna('Unknown')

Let's see again all the years we have in `year_added` column:

In [35]:
netflix_data['year_added'].unique()

array(['2021', '2020', '2019', '2018', '2017', '2016', '2015', '2014',
       '2013', '2012', '2011', '2009', '2008', 'Unknown', '2010'],
      dtype=object)

Now, before proceeding let's have a fresh look into our data in `netflix_data`:

In [36]:
# Print a sample of data for netflix_data
netflix_data.sample(n=10, random_state=100)

show_id     type                           title  \
3695   s3696  TV Show                Rabbids Invasion   
7338   s7339  TV Show  Los 10 años de Peter Capusotto   
6939   s6940    Movie      Haunting on Fraternity Row   
6626   s6627    Movie     Don't Be Afraid of the Dark   
3092   s3093  TV Show  Kevin Hart: Don’t F**k This Up   
5696   s5697    Movie                 1000 Rupee Note   
6472   s6473    Movie                      Chittagong   
4922   s4923    Movie                      Mercury 13   
4180   s4181    Movie                Old Lord Savanna   
684     s685  TV Show                           Elite   

                          director  \
3695                       Unknown   
7338                       Unknown   
6939                  Brant Sersen   
6626                    Troy Nixey   
3092                       Unknown   
5696                Shrihari Sathe   
6472                Bedabrata Pain   
4922  David Sington, Heather Walsh   
4180                  André D'Elia   
684                        Unknown   

                                                   cast  \
3695         Damien Laquet, David Gasman, Barbara Scaff   
7338                      Diego Capusotto, Ivana Acosta   
6939  Jacob Artist, Jayson Blair, Shanley Caswell, C...   
6626  Katie Holmes, Guy Pearce, Bailee Madison, Jack...   
3092                                            Unknown   
5696  Usha Naik, Sandeep Pathak, Shrikant Yadav, Gan...   
6472  Manoj Bajpayee, Barry John, Delzad Hiwale, Veg...   
4922                                            Unknown   
4180                                   Leonardo Ribeiro   
684   Danna Paola, Miguel Herrán, María Pedraza, Itz...   

                               country date_added  release_year rating  \
3695                            France 2019-07-01          2018  TV-Y7   
7338                           Unknown 2018-06-29          2015  TV-MA   
6939                     United States 2019-03-01          2018  TV-MA   
6626  United States, Australia, Mexico 2019-11-02          2010      R   
3092                     United States 2019-12-27          2019  TV-MA   
5696                             India 2016-12-01          2014  TV-14   
6472  United States, India, Bangladesh 2018-01-01          2012     NR   
4922                     United States 2018-04-20          2018  TV-PG   
4180                            Brazil 2019-01-18          2018  TV-14   
684                              Spain 2021-06-18          2021  TV-MA   

       duration                                          listed_in  \
3695   1 Season                              Kids' TV, TV Comedies   
7338   1 Season  International TV Shows, Spanish-Language TV Sh...   
6939     99 min                  Horror Movies, Independent Movies   
6626     99 min                                      Horror Movies   
3092   1 Season                                         Docuseries   
5696     89 min                       Dramas, International Movies   
6472    105 min   Dramas, Independent Movies, International Movies   
4922     79 min                                      Documentaries   
4180     96 min                Documentaries, International Movies   
684   4 Seasons  Crime TV Shows, International TV Shows, Spanis...   

                                            description year_added  
3695  Giant, rambunctious rabbits have invaded and a...       2019  
7338  Fictional personality Peter Capusotto parodies...       2018  
6939  When a fraternity's last big luau serves up ho...       2019  
6626  Young Sally Hurst discovers she isn't alone in...       2019  
3092  Amid turmoil in his career and marriage, comed...       2019  
5696  After randomly receiving a handsome political ...       2016  
6472  In the turbulent 1930s of British colonial Ind...       2018  
4922  After rigorous testing in 1961, a small group ...       2018  
4180  This documentary captures the environmental an...       2019  
684   When three working-class t

Let's see if we have any missing values in `duration` column in `movies_data` dataframe: 

In [37]:
netflix_data['duration'].isna().sum()

0

We don't have any missing values in the `duration` column.

Awesome! So, now our data is ready for analysis. Let's save the above clean data into seperate file:

In [38]:
netflix_data.to_csv('../netflix_clean_data.csv', index=False)

## Overall trend of content available on Netflix  <a id='overall-trend-of-content'></a>  
[Back to Contents](#contents)

Let's create a histogram to visualize the distribution of movies and TV shows based on their release year. This will help us understand the overall trend of content available on Netflix and identify any patterns or changes over time.

In [39]:
fig = px.histogram(
    netflix_data,
    title='Distribution of Movies and TV shows based on their release year',
    x='release_year',
    color="type",
    barmode='overlay')
fig.update_layout(yaxis_title="Number of Releases")
fig.update_layout(xaxis_title="Year of Release")
fig.show()

Let's also compare the above histogram with the number of movies released per `release_year` for both the `type`s:

In [40]:
netflix_year_wise_distribution = netflix_data.pivot_table(index='release_year', columns='type', aggfunc='count', values='show_id')
netflix_year_wise_distribution = netflix_year_wise_distribution.reset_index()
netflix_year_wise_distribution.columns = ['release_year', 'no_of_movies', 'no_of_tv_shows']

In [41]:
netflix_year_wise_distribution.head(20)

release_year  no_of_movies  no_of_tv_shows
0           1925           NaN             1.0
1           1942           2.0             NaN
2           1943           3.0             NaN
3           1944           3.0             NaN
4           1945           3.0             1.0
5           1946           1.0             1.0
6           1947           1.0             NaN
7           1954           2.0             NaN
8           1955           3.0             NaN
9           1956           2.0             NaN
10          1958           3.0             NaN
11          1959           1.0             NaN
12          1960           4.0             NaN
13          1961           1.0             NaN
14          1962           3.0             NaN
15          1963           1.0             1.0
16          1964           2.0             NaN
17          1965           2.0             NaN
18          1966           1.0             NaN
19          1967           4.0             1.0

In [42]:
netflix_year_wise_distribution.tail(20)

release_year  no_of_movies  no_of_tv_shows
54          2002          44.0             7.0
55          2003          51.0            10.0
56          2004          55.0             9.0
57          2005          67.0            13.0
58          2006          82.0            14.0
59          2007          74.0            14.0
60          2008         113.0            23.0
61          2009         118.0            34.0
62          2010         154.0            40.0
63          2011         145.0            40.0
64          2012         173.0            64.0
65          2013         225.0            63.0
66          2014         264.0            88.0
67          2015         398.0           162.0
68          2016         658.0           244.0
69          2017         767.0           265.0
70          2018         767.0           380.0
71          2019         633.0           397.0
72          2020         517.0           436.0
73          2021         277.0           315.0

Based on the data above and the insights from the histogram, **we can make several observations about the overall trend of content available on Netflix and identify some patterns or changes over time**:

1. **Increase in Content**: **The number of movies and TV shows available on Netflix has shown a significant increase over the years**. The early years (1925-1967) had a relatively smaller number of releases, but from the early 2000s onwards, there has been a substantial growth in the content offerings.

2. **Shifting Focus to TV Shows**: Initially, Netflix primarily focused on movies, as seen in the early years of the dataset where there were more movies than TV shows. However, starting from around 2013, the number of TV shows being added to Netflix started to increase rapidly.

3. **Steady Growth**: From the early 2000s to 2017, there was a consistent growth in both movie and TV show releases. The number of additions per year gradually increased, reflecting Netflix's expansion and efforts to diversify its content library.

4. **Plateauing of Movie Releases**: While the number of TV show releases has continued to rise, the number of movies added to Netflix seems to have plateaued since around 2017. **This suggests that Netflix has been focusing more on original TV show productions and acquiring TV show licenses, possibly due to their popularity and binge-watching nature**.

5. **Impact of the COVID-19 Pandemic**: Looking at the years 2020 and 2021, there is a slight decrease in the number of movie releases compared to previous years. **This could be attributed to the disruption caused by the COVID-19 pandemic, which may have affected the production and release schedules of movies**.

6. The maximum number of TV Shows were released in the year 2020 - 436 shows and the maximum number of movies were released in the years 2017 and 2018 - 767 movies.

Overall, the data indicates a significant shift in the content available on Netflix over time, with a emphasis on TV shows and a consistent growth in the number of offerings. Understanding these trends can help Netflix make informed decisions about their content acquisition and production strategies and cater to the evolving preferences of their global subscriber base.

## Study popularity of different rating categories among Netflix subscribers  <a id='study-popularity'></a>  
[Back to Contents](#contents)

 Let's plot a histogram to analyze the distribution of movies and TV shows based on their ratings. This can help us understand the popularity of different rating categories among Netflix subscribers.

In [43]:
fig = px.histogram(
    netflix_data,
    title='Distribution of movies and TV shows based on their ratings',
    x='rating',
    color="type",
    barmode='overlay')
fig.update_layout(yaxis_title="Number of movies / TV Shows")
fig.update_layout(xaxis_title="TV Ratings")
fig.show()

Let's also compare the above histogram with the number of movies released per `rating` for both the `type`s:

In [44]:
netflix_rating_wise_distribution = netflix_data.pivot_table(index='rating', columns='type', aggfunc='count', values='show_id')
netflix_rating_wise_distribution = netflix_rating_wise_distribution.reset_index()
netflix_rating_wise_distribution.columns = ['rating', 'no_of_movies', 'no_of_tv_shows']

In [45]:
netflix_rating_wise_distribution.head(10)

rating  no_of_movies  no_of_tv_shows
0      G          41.0             NaN
1  NC-17           3.0             NaN
2     NR          75.0             5.0
3     PG         287.0             NaN
4  PG-13         490.0             NaN
5      R         797.0             2.0
6  TV-14        1427.0           733.0
7   TV-G         126.0            94.0
8  TV-MA        2062.0          1145.0
9  TV-PG         540.0           323.0

Based on the data and the histogram above, **we can conclude the following about the popularity of different rating categories among Netflix subscribers**:

1. **TV-MA (Mature Audience)**: TV-MA rated content has the highest number of both movies (2062) and TV shows (1145) available on Netflix. This rating category is intended for mature audiences and indicates that **Netflix has a significant amount of content targeting adult viewers**.  

2. **TV-14 (Parents Strongly Cautioned)**: The TV-14 rating category is the second most popular among Netflix subscribers, with 1427 movies and 733 TV shows falling under this rating. This suggests that **there is a substantial demand for content suitable for viewers aged 14 and above**.

3. **R-Rated Movies**: Among movies specifically, the R rating category is one of the most prevalent, with 797 movies available on Netflix. The R rating indicates that the content may contain adult themes, strong language, violence, or other mature content. This suggests that **there is a significant audience for more mature and adult-oriented movies on the platform**.

4. **TV-PG and TV-G**: **TV-PG and TV-G ratings have a moderate presence on Netflix**, with 540 movies and 323 TV shows falling under the TV-PG category, and 126 movies and 94 TV shows classified as TV-G. These ratings are suitable for general audiences and may appeal to families and younger viewers.

5. **Limited Availability of Other Ratings**: The dataset indicates a limited number of movies and TV shows in other rating categories such as G, NC-17, NR, PG, and PG-13. This could suggest that **Netflix focuses more on content for mature audiences (TV-MA and TV-14) and may have a relatively smaller selection of content for younger or family-oriented audiences**.  

Overall, the data suggests that **Netflix subscribers have a higher preference for content rated TV-MA and TV-14, indicating a significant demand for mature and adult-oriented programming**.

## Study the growth and expansion of the platform's library <a id='study-the-growth'></a>  
[Back to Contents](#contents)

Let's create a histogram to visualize the frequency of content additions to Netflix over time. This can provide us insights into the growth and expansion of the platform's library.

In [46]:
fig = px.histogram(
    netflix_data,
    title='Frequency of content additions to Netflix over time',
    x='year_added',
    color="type",
    barmode='overlay')
fig.update_layout(yaxis_title="Number of Movies / TV Show")
fig.update_layout(xaxis_title="Year added to Netflix")
fig.show()

Let's also compare the above histogram with the number of movies/ TV shows added to Netflix per year for both the types:

In [47]:
netflix_added_year_wise_distribution = netflix_data.pivot_table(index='year_added', columns='type', aggfunc='count', values='show_id')
netflix_added_year_wise_distribution = netflix_added_year_wise_distribution.reset_index()
netflix_added_year_wise_distribution.columns = ['year_added', 'no_of_movies', 'no_of_tv_shows']

In [48]:
netflix_added_year_wise_distribution

year_added  no_of_movies  no_of_tv_shows
0        2008           1.0             1.0
1        2009           2.0             NaN
2        2010           1.0             NaN
3        2011          13.0             NaN
4        2012           3.0             NaN
5        2013           6.0             5.0
6        2014          19.0             5.0
7        2015          56.0            26.0
8        2016         253.0           176.0
9        2017         839.0           349.0
10       2018        1237.0           412.0
11       2019        1424.0           592.0
12       2020        1284.0           595.0
13       2021         993.0           505.0
14    Unknown           NaN            10.0

Based on the data above and histogram, **we can gain the following insights into the growth and expansion of Netflix's library over time**:

1. **Steady Growth**: **The number of content additions to Netflix has shown a consistent upward trend from 2008 to 2021, indicating the platform's continuous expansion**. The number of movies and TV shows added per year has generally increased over time.

2. **Accelerated Growth in Recent Years**: **The most substantial growth in content additions occurred in the years 2016 to 2021**. During this period, there was a significant surge in both movie and TV show additions, with notable spikes in 2017, 2018, and 2019.

3. **Shift towards TV Shows**: While the number of movie additions has remained relatively consistent over the years, there has been a notable increase in TV show additions. **This indicates a strategic shift by Netflix towards expanding its TV show library, possibly in response to the growing popularity of binge-watching and serialized content.**

4. **High Growth in 2017-2019**: The years 2017, 2018, and 2019 witnessed remarkable growth in content additions, with the highest number of movies and TV shows added during this period. This could be attributed to **Netflix's aggressive content acquisition and production strategies, including the release of original series and licensing deals with various studios**.

5. **Slight Decrease in 2020-2021**: The years 2020 and 2021 saw a slight decrease in the number of content additions compared to the peak years of 2017-2019.   

Overall, **the data suggests that Netflix has experienced significant growth and expansion in its content library over the years, with a focus on increasing the number of TV shows. The platform has continuously added new movies and TV shows, contributing to its vast collection and offering diverse options to its subscribers**.

## Analyze the duration against the release year to check for any patterns or changes in the length of content over time

In order to analyze the duration (in minutes for movies or number of seasons for TV shows) against the release year can reveal any patterns or changes in the length of content over time, it will be great if we could divide the dataframe - `netflix_data` into two parts - `netflix_movies` and `netflix_tv_shows`:

In [49]:
netflix_movies = netflix_data[netflix_data['type'] == 'Movie']
netflix_movies = netflix_movies.reset_index(drop=True)
netflix_movies.head(5)

show_id   type                             title  \
0      s1  Movie              Dick Johnson Is Dead   
1      s7  Movie  My Little Pony: A New Generation   
2      s8  Movie                           Sankofa   
3     s10  Movie                      The Starling   
4     s13  Movie                      Je Suis Karl   

                        director  \
0                Kirsten Johnson   
1  Robert Cullen, José Luis Ucha   
2                   Haile Gerima   
3                 Theodore Melfi   
4            Christian Schwochow   

                                                cast  \
0                                            Unknown   
1  Vanessa Hudgens, Kimiko Glenn, James Marsden, ...   
2  Kofi Ghanaba, Oyafunmike Ogunlano, Alexandra D...   
3  Melissa McCarthy, Chris O'Dowd, Kevin Kline, T...   
4  Luna Wedler, Jannis Niewöhner, Milan Peschel, ...   

                                             country date_added  release_year  \
0                                      United States 2021-09-25          2020   
1                                            Unknown 2021-09-24          2021   
2  United States, Ghana, Burkina Faso, United Kin... 2021-09-24          1993   
3                                      United States 2021-09-24          2021   
4                            Germany, Czech Republic 2021-09-23          2021   

  rating duration                                         listed_in  \
0  PG-13   90 min                                     Documentaries   
1     PG   91 min                          Children & Family Movies   
2  TV-MA  125 min  Dramas, Independent Movies, International Movies   
3  PG-13  104 min                                  Comedies, Dramas   
4  TV-MA  127 min                      Dramas, International Movies   

                                         description year_added  
0  As her father nears the end of his life, filmm...       2021  
1  Equestria's divided. But a bright-eyed hero be...       2021  
2  On a photo shoot in Ghana, an American model s...       2021  
3  A woman adjusting to life after a loss contend...       2021  
4  After most of her family is murdered in a terr...       2021

In [50]:
netflix_tv_shows = netflix_data[netflix_data['type'] == 'TV Show']
netflix_tv_shows = netflix_tv_shows.reset_index(drop=True)
netflix_tv_shows.head(5)

show_id     type                  title         director  \
0      s2  TV Show          Blood & Water          Unknown   
1      s3  TV Show              Ganglands  Julien Leclercq   
2      s4  TV Show  Jailbirds New Orleans          Unknown   
3      s5  TV Show           Kota Factory          Unknown   
4      s6  TV Show          Midnight Mass    Mike Flanagan   

                                                cast       country date_added  \
0  Ama Qamata, Khosi Ngema, Gail Mabalane, Thaban...  South Africa 2021-09-24   
1  Sami Bouajila, Tracy Gotoas, Samuel Jouy, Nabi...       Unknown 2021-09-24   
2                                            Unknown       Unknown 2021-09-24   
3  Mayur More, Jitendra Kumar, Ranjan Raj, Alam K...         India 2021-09-24   
4  Kate Siegel, Zach Gilford, Hamish Linklater, H...       Unknown 2021-09-24   

   release_year rating   duration  \
0          2021  TV-MA  2 Seasons   
1          2021  TV-MA   1 Season   
2          2021  TV-MA   1 Season   
3          2021  TV-MA  2 Seasons   
4          2021  TV-MA   1 Season   

                                           listed_in  \
0    International TV Shows, TV Dramas, TV Mysteries   
1  Crime TV Shows, International TV Shows, TV Act...   
2                             Docuseries, Reality TV   
3  International TV Shows, Romantic TV Shows, TV ...   
4                 TV Dramas, TV Horror, TV Mysteries   

                                         description year_added  
0  After crossing paths at a party, a Cape Town t...       2021  
1  To protect his family from a powerful drug lor...       2021  
2  Feuds, flirtations and toilet talk go down amo...       2021  
3  In a city of coaching centers known to train I...       2021  
4  The arrival of a charismatic young priest brin...       2021

Let's clean up the `duration` column in both the dataframes - `netflix_movies` and `netflix_tv_shows`:

In [51]:
netflix_movies['duration'] = netflix_movies['duration'].str.replace(' min', '')
netflix_movies['duration'] = netflix_movies['duration'].astype(int)
netflix_movies.head(5)

show_id   type                             title  \
0      s1  Movie              Dick Johnson Is Dead   
1      s7  Movie  My Little Pony: A New Generation   
2      s8  Movie                           Sankofa   
3     s10  Movie                      The Starling   
4     s13  Movie                      Je Suis Karl   

                        director  \
0                Kirsten Johnson   
1  Robert Cullen, José Luis Ucha   
2                   Haile Gerima   
3                 Theodore Melfi   
4            Christian Schwochow   

                                                cast  \
0                                            Unknown   
1  Vanessa Hudgens, Kimiko Glenn, James Marsden, ...   
2  Kofi Ghanaba, Oyafunmike Ogunlano, Alexandra D...   
3  Melissa McCarthy, Chris O'Dowd, Kevin Kline, T...   
4  Luna Wedler, Jannis Niewöhner, Milan Peschel, ...   

                                             country date_added  release_year  \
0                                      United States 2021-09-25          2020   
1                                            Unknown 2021-09-24          2021   
2  United States, Ghana, Burkina Faso, United Kin... 2021-09-24          1993   
3                                      United States 2021-09-24          2021   
4                            Germany, Czech Republic 2021-09-23          2021   

  rating  duration                                         listed_in  \
0  PG-13        90                                     Documentaries   
1     PG        91                          Children & Family Movies   
2  TV-MA       125  Dramas, Independent Movies, International Movies   
3  PG-13       104                                  Comedies, Dramas   
4  TV-MA       127                      Dramas, International Movies   

                                         description year_added  
0  As her father nears the end of his life, filmm...       2021  
1  Equestria's divided. But a bright-eyed hero be...       2021  
2  On a photo shoot in Ghana, an American model s...       2021  
3  A woman adjusting to life after a loss contend...       2021  
4  After most of her family is murdered in a terr...       2021

In [52]:
netflix_tv_shows['duration'] = netflix_tv_shows['duration'].str.replace(' Season', '')
netflix_tv_shows['duration'] = netflix_tv_shows['duration'].str.replace('s', '')
netflix_tv_shows['duration'] = netflix_tv_shows['duration'].astype(int)
netflix_tv_shows.head(5)

show_id     type                  title         director  \
0      s2  TV Show          Blood & Water          Unknown   
1      s3  TV Show              Ganglands  Julien Leclercq   
2      s4  TV Show  Jailbirds New Orleans          Unknown   
3      s5  TV Show           Kota Factory          Unknown   
4      s6  TV Show          Midnight Mass    Mike Flanagan   

                                                cast       country date_added  \
0  Ama Qamata, Khosi Ngema, Gail Mabalane, Thaban...  South Africa 2021-09-24   
1  Sami Bouajila, Tracy Gotoas, Samuel Jouy, Nabi...       Unknown 2021-09-24   
2                                            Unknown       Unknown 2021-09-24   
3  Mayur More, Jitendra Kumar, Ranjan Raj, Alam K...         India 2021-09-24   
4  Kate Siegel, Zach Gilford, Hamish Linklater, H...       Unknown 2021-09-24   

   release_year rating  duration  \
0          2021  TV-MA         2   
1          2021  TV-MA         1   
2          2021  TV-MA         1   
3          2021  TV-MA         2   
4          2021  TV-MA         1   

                                           listed_in  \
0    International TV Shows, TV Dramas, TV Mysteries   
1  Crime TV Shows, International TV Shows, TV Act...   
2                             Docuseries, Reality TV   
3  International TV Shows, Romantic TV Shows, TV ...   
4                 TV Dramas, TV Horror, TV Mysteries   

                                         description year_added  
0  After crossing paths at a party, a Cape Town t...       2021  
1  To protect his family from a powerful drug lor...       2021  
2  Feuds, flirtations and toilet talk go down amo...       2021  
3  In a city of coaching centers known to train I...       2021  
4  The arrival of a charismatic young priest brin...       2021

Let's plot a scatterplot for movies on Netflix - `release_year` on x-axis and `duration` on y-axis. This plot can help determine if there is a trend towards shorter or longer movies.

In [53]:
fig = px.scatter(
    netflix_movies,
    title='Distribution of movies duration against the release year',
    x='release_year',
    y='duration',
    height=800)
fig.update_layout(yaxis_title="Duration (in mins)")
fig.update_layout(xaxis_title="Year of Release")
fig.show()

Let's plot a scatterplot for TV Shows on Netflix - `release_year` on x-axis and `duration` on y-axis. This plot can help determine if there is a trend towards shorter or longer TV Shows.

In [54]:
fig = px.scatter(
    netflix_tv_shows,
    title='Distribution of TV Shows duration against the release year',
    x='release_year',
    y='duration',
    height=800)
fig.update_layout(yaxis_title="Duration (no of seasons)")
fig.update_layout(xaxis_title="Year of Release")
fig.show()

With the scatter plots above, we can conclude two things:
1. There is a trend towards shorter and medium length movies.
2. There are a few really long length movies released lately too.
3. There is a trend towards longer TV Shows.